radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/messaging/MessageCentralImpl.java

```diff
@@ -163,6 +169,11 @@
 	}
 
 	private Optional<MessageFromPeer<Message>> processInboundMessage(InboundMessage inboundMessage) {
+		final var messageQueuedTime = System.currentTimeMillis() - inboundMessage.receiveTime();
+		avgMessageQueuedTime.update(messageQueuedTime);
+		totalMessageQueuedTime = Math.max(totalMessageQueuedTime + messageQueuedTime, 0L);
+		updateCounters();
```

Comment:
Please, use `Stopwatch` to measure time.

Reply:
I wouldn't like to introduce a stateful property (which as far as I see Stopwatch is) to InboundMessage

Reply:
I've used it for measuring the processing time below.

Reply:
This is a good source about the topic https://www.baeldung.com/java-measure-elapsed-time